query_optimizer/expressions/ExpressionUtil.cpp

```diff
@@ -29,7 +33,20 @@ AttributeReferencePtr ToRef(const NamedExpressionPtr &expression) {
                                     expression->attribute_name(),
                                     expression->attribute_alias(),
                                     expression->relation_name(),
-                                    expression->getValueType());
+                                    expression->getValueType(),
+                                    AttributeReferenceScope::kLocal);
+}
+
+std::vector<AttributeReferencePtr> GetAttributeReferencesWithinScope(
+    const std::vector<AttributeReferencePtr> &attributes,
+    const AttributeReferenceScope scope) {
+  std::vector<AttributeReferencePtr> filtered_attributes;
+  for (const auto& attr_it : attributes) {
```

Comment:
If we apply the reference sigil standard here, it would look like:

`for (const auto &attr_it : attributes) {`.